### PR TITLE
Feature pagination

### PR DESCRIPTION
**Pull request by Abhishek-Raj-AT**
Summary:
Add pagination functionality
ChangeLog:
Added pagination functionality
Test cases:
- [ ] Check the pagination functionality code
- [ ] Check the pagination functionality in film, people, planets 
- [ ] Check previous and next button is working or not
- [ ] when we click on next button and then go to the next page
- [ ] when we click on prev button and then go to the previous page

### CODE REVIEW

File: src/redux/FilmSlice/FilmType.tsx [hunk 1]
```diff
@@ -16,5 +16,9 @@ export interface Film {
 export interface FilmList {
   list: Film[];
   isLoading: boolean;
-  value: number;
+  page: number;
+  total: number;
+  nextPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/FilmSlice/FilmType.tsx]: Please explain me the logic behind this ?

File: src/redux/FilmSlice/FilmType.tsx [hunk 2]
```diff
@@ -16,5 +16,9 @@ export interface Film {
...
+  prevPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/FilmSlice/FilmType.tsx]: Please explain me the logic behind this ?

File: src/redux/FilmSlice/index.tsx [hunk 3]
```diff
@@ -1,18 +1,26 @@
...
 import { FilmList } from "./FilmType";
 import { createSlice } from "@reduxjs/toolkit";
 
 const initialState: FilmList = {
   list: [],
   isLoading: false,
-  value: 5,
+  page: constant.page.defaultNumber,
+  total: constant.page.defaultTotal,
+  nextPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/FilmSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/FilmSlice/index.tsx [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
+  prevPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/FilmSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/PeopleSlice/PeopleType.tsx [hunk 5]
```diff
@@ -14,4 +14,9 @@ export interface Peoples {
 export interface PeoplesList {
   list: Peoples[];
   isLoading: boolean;
+  page: number;
+  total: number;
+  nextPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/PeopleSlice/PeopleType.tsx]: Please explain me the logic behind this ?

File: src/redux/PeopleSlice/PeopleType.tsx [hunk 6]
```diff
@@ -14,4 +14,9 @@ export interface Peoples {
...
+  prevPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/PeopleSlice/PeopleType.tsx]: Please explain me the logic behind this ?

File: src/redux/PeopleSlice/index.tsx [hunk 7]
```diff
@@ -1,15 +1,25 @@
...
 import { PeoplesList } from "./PeopleType";
 import { getPeopleActions } from "./PeopleAsyncThunk";
+import constant from "../../config/constant";
 
 const initialState: PeoplesList = {
   list: [],
   isLoading: false,
+  page: constant.page.defaultNumber,
+  total: constant.page.defaultTotal,
+  nextPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/PeopleSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/PeopleSlice/index.tsx [hunk 8]
```diff
@@ -1,15 +1,25 @@
...
+  prevPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/PeopleSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/PlanetsSlice/PlanetTypes.tsx [hunk 9]
```diff
@@ -17,4 +17,9 @@ export interface Planet {
 export interface PlanetList {
   list: Planet[];
   isLoading: boolean;
+  page: number;
+  total: number;
+  nextPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/PlanetsSlice/PlanetTypes.tsx]: Please explain me the logic behind this ?

File: src/redux/PlanetsSlice/PlanetTypes.tsx [hunk 10]
```diff
@@ -17,4 +17,9 @@ export interface Planet {
...
+  prevPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/PlanetsSlice/PlanetTypes.tsx]: Please explain me the logic behind this ?

File: src/redux/PlanetsSlice/index.tsx [hunk 11]
```diff
@@ -1,23 +1,34 @@
...
 import { PlanetList } from "./PlanetTypes";
 import { getPlanetActions } from "./PlanetAsyncThunk";
+import constant from "../../config/constant";
 
 const initialState: PlanetList = {
   list: [],
   isLoading: false,
+  page: constant.page.defaultNumber,
+  total: constant.page.defaultTotal,
+  nextPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/PlanetsSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/PlanetsSlice/index.tsx [hunk 12]
```diff
@@ -1,23 +1,34 @@
...
+  prevPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/PlanetsSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/SpeciesSlice/SpeciesTypes.tsx [hunk 13]
```diff
@@ -18,4 +18,9 @@ export interface Species {
 export interface SpeciesList {
   list: Species[]
   isLoading: boolean;
+  page: number;
+  total: number;
+  nextPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/SpeciesSlice/SpeciesTypes.tsx]: Please explain me the logic behind this ?

File: src/redux/SpeciesSlice/SpeciesTypes.tsx [hunk 14]
```diff
@@ -18,4 +18,9 @@ export interface Species {
...
+  prevPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/SpeciesSlice/SpeciesTypes.tsx]: Please explain me the logic behind this ?

File: src/redux/SpeciesSlice/index.tsx [hunk 15]
```diff
@@ -1,15 +1,25 @@
...
 import { getSpeciesActions } from "./SpeciesAsyncThunk";
 import { SpeciesList } from "./SpeciesTypes";
+import constant from "../../config/constant";
 
 const initialState: SpeciesList = {
   list: [],
   isLoading: false,
+  page: constant.page.defaultNumber,
+  total: constant.page.defaultTotal,
+  nextPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/SpeciesSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/SpeciesSlice/index.tsx [hunk 16]
```diff
@@ -1,15 +1,25 @@
...
+  prevPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/SpeciesSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/StarshipSlice/StarshipTypes.tsx [hunk 17]
```diff
@@ -21,4 +21,9 @@ export interface Starship {
 export interface StarshipList {
   list: Starship[];
   isLoading: boolean;
+  page: number;
+  total: number;
+  nextPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/StarshipSlice/StarshipTypes.tsx]: Please explain me the logic behind this ?

File: src/redux/StarshipSlice/StarshipTypes.tsx [hunk 18]
```diff
@@ -21,4 +21,9 @@ export interface Starship {
...
+  prevPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/StarshipSlice/StarshipTypes.tsx]: Please explain me the logic behind this ?

File: src/redux/StarshipSlice/index.tsx [hunk 19]
```diff
@@ -1,15 +1,25 @@
...
 import { StarshipList } from "./StarshipTypes";
 import { getStarshipActions } from "./StarshipAsyncThunk";
+import constant from "../../config/constant";
 
 const initialState: StarshipList = {
   list: [],
   isLoading: false,
+  page: constant.page.defaultNumber,
+  total: constant.page.defaultTotal,
+  nextPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/StarshipSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/StarshipSlice/index.tsx [hunk 20]
```diff
@@ -1,15 +1,25 @@
...
+  prevPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/StarshipSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/VehicleSlice/Vehicletypes.tsx [hunk 21]
```diff
@@ -19,4 +19,9 @@ export interface Vehicle {
 export interface VehicleList {
   list: Vehicle[];
   isLoading: boolean;
+  page: number;
+  total: number;
+  nextPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/VehicleSlice/Vehicletypes.tsx]: Please explain me the logic behind this ?

File: src/redux/VehicleSlice/Vehicletypes.tsx [hunk 22]
```diff
@@ -19,4 +19,9 @@ export interface Vehicle {
...
+  prevPageUrl: null | string;
```
Comment (bhavikg-dev) [inline src/redux/VehicleSlice/Vehicletypes.tsx]: Please explain me the logic behind this ?

File: src/redux/VehicleSlice/index.tsx [hunk 23]
```diff
@@ -1,15 +1,25 @@
...
 import { VehicleList } from "./Vehicletypes";
 import { getVehicleActions } from "./VehicleAsyncThunk";
+import constant from "../../config/constant";
 
 const initialState: VehicleList = {
   list: [],
   isLoading: false,
+  page: constant.page.defaultNumber,
+  total: constant.page.defaultTotal,
+  nextPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/VehicleSlice/index.tsx]: Please explain me the logic behind this ?

File: src/redux/VehicleSlice/index.tsx [hunk 24]
```diff
@@ -1,15 +1,25 @@
...
+  prevPageUrl: null,
```
Comment (bhavikg-dev) [inline src/redux/VehicleSlice/index.tsx]: Please explain me the logic behind this ?

**Review by bhavikg-dev — CHANGES_REQUESTED** [review]

File: src/redux/FilmSlice/FilmType.tsx (hunk 1 above)
Comment (Abhishek-Raj-AT) [reply src/redux/FilmSlice/FilmType.tsx]: Sorry sir, I removed this issue

File: src/redux/FilmSlice/FilmType.tsx (hunk 2 above)
Comment (Abhishek-Raj-AT) [reply src/redux/FilmSlice/FilmType.tsx]: I forgot to remove, I removed this issue

File: src/redux/PeopleSlice/PeopleType.tsx (hunk 5 above)
Comment (Abhishek-Raj-AT) [reply src/redux/PeopleSlice/PeopleType.tsx]: I removed this issue

File: src/redux/VehicleSlice/index.tsx (hunk 24 above)
Comment (Abhishek-Raj-AT) [reply src/redux/VehicleSlice/index.tsx]: I removed All the issue

**Review by bhavikg-dev — APPROVED** [review]
Approving though some issues are still to be addressed

**Pull request merged**